src/de/teamgrit/grit/entities/Exercise.java

```diff
@@ -70,552 +68,864 @@
+	}
+
+	/**
+	 * Gets the ID of the Exercise.
+	 * 
+	 * @return the id
+	 */
+	public int getId() {
+		return id;
+	}
+
+	/**
+	 * returns the exercise context.
+	 * 
+	 * @return the exercise context
+	 */
+	protected ExerciseContext getContext() {
+		return context;
+	}
+
+	// --------------------- HELPER METHODS ---------------------
+
+	/**
+	 * Cleanup when stopping a task. Delete all generated files.
+	 */
+	private void cleanup() {
+		// Delete binaries, test files, temporary files and fetched sources.
+		try {
+		  LOGGER.info("Cleaning up exercise "+context.getExerciseName());
+			FileUtils.deleteDirectory(context.getBinPath().toFile());
+			FileUtils.deleteDirectory(context.getTempPdfPath().toFile());
+			FileUtils.deleteDirectory(context.getFetchPath().toFile());
+			FileUtils.deleteDirectory(context.getTests().toFile());
+
+			// Deleting auxiliary files for report.pdf generation
+			File f = context.getOutputPath().toFile();
+			File[] paths = f.listFiles();
+
+			for (File file : paths) {
+				String temp = file.toString();
+				temp = temp.substring(temp.lastIndexOf("/") + 1);
+				if (!(temp.equals("report.pdf")))
+					file.delete();
+			}
+
+		} catch (NullPointerException e) {
+			LOGGER.severe("Error while trying to clean up: "
+					+ e.toString());
+		} catch (IOException e) {
+			LOGGER.severe("Error while trying to clean up: " + e.toString());
+		}
+	}
+
+	/**
+	 * getCompilerError returns the compiler errors, so that they can be send to
+	 * the students in the method notifyStudentsWithCorruptedSubmission()
+	 * 
+	 * @param sub
+	 *          The submission of the student with compiler errors
+     * @return
+     *          Returns the compilerOutput as a String
+	 */
+	private String getCompilerError(Submission sub) {
+		// The following code creates a string with the compilerError.
+		String compilerOutput = "Die Abgabe kompiliert nicht." + "\n\n";
+		for (String error : sub.getCheckingResult().getCompilerOutput()
+				.getCompilerErrors()) {
+			compilerOutput += "Compiler Errors:" + "\n\n";
+			compilerOutput += error + "\n";
+			compilerOutput += "\n\n";
+		}
+
+        for (String warning : sub.getCheckingResult()
+                .getCompilerOutput().getCompilerWarnings()) {
+			compilerOutput += "Compiler Warnings:" + "\n\n";
+			compilerOutput += warning + "\n";
+			compilerOutput += "\n\n";
+		}
+
+		for (String info : sub.getCheckingResult().getCompilerOutput()
+				.getCompilerInfos()) {
+
+			compilerOutput += "Compiler Infos:" + "\n\n";
+			compilerOutput += info + "\n";
+			compilerOutput += "\n\n";
+		}
+		return compilerOutput;
+	}
+
+	/**
+	 * Sends an E-Mail to every student who has send a solution as an email.
+	 * 
+	 * @param submissions
+	 *            Every new submission which is received by the email-server.
+     * 			  List of submissions which contain all relevant
+     *            data in order to send mails
+	 */
+    private void sendReceivingNotification(
```

Comment:
Explizietes Mailhaindling gehört in eine eigene Klasse!
``Exercise`` ist nicht für solche details!